packages/xstate-svelte/src/useMachine.ts

```diff
@@ -1,40 +1,71 @@
-import { readable } from 'svelte/store';
+import { Readable, readable } from 'svelte/store';
 import {
-  interpret,
+  AnyStateMachine,
+  AreAllImplementationsAssumedToBeProvided,
   EventObject,
-  StateMachine,
-  State,
+  InternalMachineOptions,
+  interpret,
+  InterpreterFrom,
   InterpreterOptions,
-  MachineOptions,
+  State,
   StateConfig,
-  Typestate
+  StateFrom
 } from 'xstate';
 
+type Prop<T, K> = K extends keyof T ? T[K] : never;
 interface UseMachineOptions<
   TContext extends object,
   TEvent extends EventObject
 > {
   /**
    * If provided, will be merged with machine's `context`.
    */
-  context: Partial<TContext>;
+  context?: Partial<TContext>;
   /**
    * The state to rehydrate the machine to. The machine will
    * start at this state instead of its `initialState`.
    */
-  state: StateConfig<TContext, TEvent>;
+  state?: StateConfig<TContext, TEvent>;
 }
 
-export function useMachine<
-  TContext extends object,
-  TEvent extends EventObject,
-  TTypestate extends Typestate<TContext>
->(
-  machine: StateMachine<TContext, any, TEvent, TTypestate>,
-  options: Partial<InterpreterOptions> &
-    Partial<UseMachineOptions<TContext, TEvent>> &
-    Partial<MachineOptions<TContext, TEvent>> = {}
-) {
+type RestParams<
+  TMachine extends AnyStateMachine
+> = AreAllImplementationsAssumedToBeProvided<
+  TMachine['__TResolvedTypesMeta']
+> extends false
+  ? [
+      options: InterpreterOptions &
+        UseMachineOptions<TMachine['__TContext'], TMachine['__TEvent']> &
+        InternalMachineOptions<
+          TMachine['__TContext'],
+          TMachine['__TEvent'],
+          TMachine['__TResolvedTypesMeta'],
+          true
+        >
+    ]
+  : [
+      options?: InterpreterOptions &
+        UseMachineOptions<TMachine['__TContext'], TMachine['__TEvent']> &
+        InternalMachineOptions<
```

Comment:
Not right now because `InternalMachineOptions` is also used within `createMachine` and `TMachine` doesn't exist there (yet). Of course, a helper type could be introduced that would be tailored for this use case here - but it's probably better to clean it up later in v5. This pattern is already used both in `@xstate/react` and `@xstate/vue`